src/types/api/auditlog.ts

```diff
@@ -0,0 +1,217 @@
+  target_id: string | null;
+  /** changes made to the target_id */
+  changes?: DiscordAuditLogChangePayload[];
+  /** the user who made the changes */
+  user_id: string;
+  /** id of the entry */
+  id: string;
+  /** type of action that occured */
+  action_type: DiscordAuditLogEvent;
+  /** additional info for certain action types */
+  options?: DiscordOptionalAuditEntryInfoParam;
+  /** the reason for the change (0-512 characters) */
+  reason?: string;
+}
+
+/** https://discord.com/developers/docs/resources/audit-log#audit-log-entry-object-audit-log-events */
+export enum DiscordAuditLogEvent {
+  GUILD_UPDATE = 1,
+  CHANNEL_CREATE = 10,
+  CHANNEL_UPDATE,
+  CHANNEL_DELETE,
+  CHANNEL_OVERWRITE_CREATE,
+  CHANNEL_OVERWRITE_UPDATE,
+  CHANNEL_OVERWRITE_DELETE,
+  MEMBER_KICK = 20,
+  MEMBER_PRUNE,
+  MEMBER_BAN_ADD,
+  MEMBER_BAN_REMOVE,
+  MEMBER_UPDATE,
+  MEMBER_ROLE_UPDATE,
+  MEMBER_MOVE,
+  MEMBER_DISCONNECT,
+  BOT_ADD,
+  ROLE_CREATE = 30,
+  ROLE_UPDATE,
+  ROLE_DELETE,
+  INVITE_CREATE = 40,
+  INVITE_UPDATE,
+  INVITE_DELETE,
+  WEBHOOK_CREATE = 50,
+  WEBHOOK_UPDATE,
+  WEBHOOK_DELETE,
+  EMOJI_CREATE = 60,
+  EMOJI_UPDATE,
+  EMOJI_DELETE,
+  MESSAGE_DELETE = 72,
+  MESSAGE_BULK_DELETE,
+  MESSAGE_PIN,
+  MESSAGE_UNPIN,
+  INTEGRATION_CREATE = 80,
+  INTEGRATION_UPDATE,
+  INTEGRATION_DELETE,
+}
+
+/** https://discord.com/developers/docs/resources/audit-log#audit-log-entry-object-optional-audit-entry-info */
+export interface DiscordOptionalAuditEntryInfoParam {
+  /** number of days after which inactive members were kicked, type: MEMBER_PRUNE */
+  delete_member_days: string;
+  /** number of members removed by the prune, type: MEMBER_PRUNE */
+  members_removed: string;
+  /** channel in which the entities were targeted, types: MEMBER_MOVE & MESSAGE_PIN & MESSAGE_UNPIN & MESSAGE_DELETE */
+  channel_id: string;
+  /** id of the message that was targeted, types: MESSAGE_PIN & MESSAGE_UNPIN */
+  message_id: string;
+  /** number of entities that were targeted, types: MESSAGE_DELETE & MESSAGE_BULK_DELETE & MEMBER_DISCONNECT & MEMBER_MOVE */
+  count: string;
+  /** id of the overwritten entity, types CHANNEL_OVERWRITE_CREATE & CHANNEL_OVERWRITE_UPDATE & CHANNEL_OVERWRITE_DELETE */
+  id: string;
+  /** type of overwritten entity - "0", for "role", or "1" for "member", types: CHANNEL_OVERWRITE_CREATE & CHANNEL_OVERWRITE_UPDATE & CHANNEL_OVERWRITE_DELETE */
+  type: string;
+  /** name of the role if type is "0" (not present if type is "1"), types: CHANNEL_OVERWRITE_CREATE & CHANNEL_OVERWRITE_UPDATE & CHANNEL_OVERWRITE_DELETE */
+  role_name: string;
+}
+
+/** https://discord.com/developers/docs/resources/audit-log#audit-log-change-object-audit-log-change-structure */
+export interface DiscordAuditLogChangePayload {
+  /** new value of the key */
+  new_value?: DiscordAuditLogChangeKeyPayload;
+  /** old value of the key */
+  old_value?: DiscordAuditLogChangeKeyPayload;
```

Comment:
These types are wrong. You can use this files as reference https://github.com/discordeno/discordeno/pull/487/files